### PR TITLE
Additions to kernel : 3D

### PR DESCRIPTION
**Pull request by RubendeBruin**
Added Line_3
Added Transform3

completed the distance_to for 2d
added intersect, dointersect and distance_to for 3d

(also includes PR for building on windows with conda envs)

### CODE REVIEW

**Comment by wolfv** [comment]
Hi @RubendeBruin this looks great! Thanks a lot for your work on this. Do you think you can fix the issue on OS X?

And just to nitpick, there is some slightly inconsistent formatting :)

**Comment by RubendeBruin** [comment]
Hi @wolfv , 

I don't think I'll be able to fix these OS X issues without help.

The errors are due to the following code in intersection_3_1_impl.h:

```
template <class K>
inline
bool
do_intersect(const typename K::Line_3  &l,
             const typename K::Segment_3  &s,
             const K & k)
{
  CGAL_precondition(! l.is_degenerate () && ! s.is_degenerate () );

  bool b=do_intersect(l,s.supporting_line(),k);           <---error: call to 'do_intersect' is ambiguous                                

  if (b)
  {
    //supporting_line intersects: points are coplanar
    typename K::Coplanar_orientation_3 cpl_orient=k.coplanar_orientation_3_object();
    typename K::Point_3 p1=l.point(0);
    typename K::Point_3 p2=l.point(1);
    ::CGAL::Orientation or1 =  cpl_orient(p1,p2,s[0]);
       
    if ( or1 == COLLINEAR ) return true;
    
    ::CGAL::Orientation or2 =  cpl_orient(p1,p2,s[1]);
    return or1!=or2;
  }
  return false;
}

template <class K>
inline
bool
do_intersect(const typename K::Segment_3  &s,
             const typename K::Line_3  &l,
             const K & k)
{

  return do_intersect(l,s,k);                                              <---error: call to 'do_intersect' is ambiguous

}

```
which is not one of the files that I changed.

I do not have OSX and the build works fine on Linux and Windows. No idea what is different in clang that makes this fail.

File: src/kernel.cpp [hunk 1]
```diff
@@ -49,6 +49,11 @@ bool do_intersect(T1 p, T2 q) {
     .def(+py::self)  \
 
 
+void info()
```
Comment (wolfv) [inline src/kernel.cpp:52]: this should be removed?

**Comment by wolfv** [comment]
Ah and if you don't mind, would you squash the commits a little bit (with git rebase)? So that we have 2 or 3 remaining commits?

File: src/kernel.cpp (hunk 1 above)
Comment (RubendeBruin) [reply src/kernel.cpp:52]: yes

**Comment by RubendeBruin** [comment]
I was still in the iterative process of enabling everything that does work in OS-X. Will squash afterwards

**Comment by wolfv** [comment]
ok, even better!

**Comment by RubendeBruin** [comment]
squashed and resubmitted as #46. Closing this request.

**Pull request closed without merging**